sdk/core/core-compat/src/shimClient.ts

```diff
@@ -0,0 +1,32 @@
+import { ServiceClient } from "@azure/core-client";
+import { KeepAliveOptions } from "./policies/keepAliveOptions";
+import { createKeepAlivePolicy } from "./policies/keepAlivePolicy";
+import { RedirectPolicyOptions } from "./policies/redirectPolicyOptions";
+import { redirectPolicy } from "@azure/core-rest-pipeline";
+import { ServiceClientOptions } from "@azure/core-client";
+
+export interface ShimClientOptions extends ServiceClientOptions {
+  keepAliveOptions?: KeepAliveOptions;
+  redirectOptions?: RedirectPolicyOptions;
+}
+
+export class ShimClient extends ServiceClient {
+  constructor(options: ShimClientOptions) {
+    super(options);
+
+    if (options.keepAliveOptions?.enable === false) {
+      this.pipeline.addPolicy(createKeepAlivePolicy());
+    }
+
+    if (options.redirectOptions?.handleRedirects === true) {
+      const maxRetries = options.redirectOptions.maxRetries
+        ? options.redirectOptions.maxRetries
+        : 20;
+      this.pipeline.addPolicy(
```

Comment:
IIRC the corev2 pipeline already has a redirect policy, so do we need to remove the existing one before adding this one?

Reply:
Modified the logic